src/Reshuffle.ts

```diff
@@ -136,13 +136,16 @@ export default class Reshuffle {
   }
 
   setPersistentStore(adapter: PersistentStoreAdapter) {
-    const ps = new PersistentStore(adapter)
-    this.registry.common.persistentStore = ps
-    return ps
+    this.registry.common.persistentStore = adapter;
+    return adapter;
   }
 
   getPersistentStore() {
-    return this.registry.common.persistentStore
+
+    if(this.registry.common.persistentStore){
+      this.registry.common.persistentStore = new MemoryStoreAdapter();
```

Comment:
Maybe use the setter here? then you can do
`return this.registry.persistentStore || setPersistentStore(new MemoryStoreAdapter());`